src/pages/iou/request/IOURequestStartPage.js

```diff
@@ -108,6 +109,12 @@ function IOURequestStartPage({
         [previousIOURequestType, reportID],
     );
 
+    if (!transaction.transactionID) {
+        // The draft transaction is initialized only after the component is mounted,
+        // which will lead to briefly displaying the Not Found page without this loader.
+        return <FullScreenLoadingIndicator />;
+    }
```

Comment:
@tgolen even though this was not in the original proposal, I've noticed that the NotFound page shows also while the modal is opening. The root cause here is different – it's caused by the draft transaction being initialized only after the component is mounted, by calling `IOU.startMoneyRequest_temporaryForRefactor`.
Please let me know if this change is not suitable here – I couldn't come up with a better fix for this case so far.

Reply:
Thanks for digging into that. I think this fix is fine and I 💚 the fix here.